### PR TITLE
Move methods from Client to Backend for pandas based backends

### PR DESCRIPTION
**Pull request by datapythonista**
xref #3017

Same as #3018, but for pandas backends (pandas, dask, csv, parquet and hdf5).

### CODE REVIEW

File: ibis/backends/csv/__init__.py
Comment (cpcloud) [inline ibis/backends/csv/__init__.py:68]: I guess this will go away after everything becomes a backend?

Comment (datapythonista) [reply ibis/backends/csv/__init__.py:68]: Yes, when client classes are deleted and fully replaced by backend classes (with some luck in the next PR after this one), all those will register only backend classes, not both.

**Comment by datapythonista** [comment]
@jreback @cpcloud this should be ready now. As in the previous PR, this moves the methods from client classes to backend classes without changes more than the required (like `self.client` to `self` to find the new location of the methods, or to update the classes in the dispatchers).

In this case, I'm already removing the client classes here. I'll follow up removing the client classes from the SQL backends, which should be easy now, and from my side we're good to release.

File: ibis/backends/pandas/__init__.py
Comment (jreback) [inline ibis/backends/pandas/__init__.py:30]: can these imports be done at the top?

File: ibis/backends/pandas/__init__.py
Comment (jreback) [inline ibis/backends/pandas/__init__.py:107]: same

**Review by jreback — CHANGES_REQUESTED** [review]

File: ibis/backends/base/file/__init__.py
Comment (datapythonista) [reply ibis/backends/base/file/__init__.py:119]: Yes, happy to work on typing stuff after the release. But I think we should first the broken mypy errors, since we're not validating types at the moment, and adding types that are not checked doesn't add that much value in my opinion.

File: ibis/backends/pandas/__init__.py
Comment (datapythonista) [reply ibis/backends/pandas/__init__.py:30]: Unfortunately, the `ibis.backends.pandas.execution` module imports `ibis.backends.pandas` to be able to register the pandas `Backend` class in some dispatchers.  This created a lot of problems in the dispatching, and I'd be happy to try to improve that code later, since it's quite difficult to maintain. But for now this is the best solution I've found.

Same applies to the code in the other comment below.

File: ibis/backends/base/file/__init__.py
Comment (jreback) [reply ibis/backends/base/file/__init__.py:119]: sure that's why i said followup

**Comment by jreback** [comment]
ok seems fine to me, @cpcloud any issues here?

@datapythonista after this anything else for release (aside from maybe cleaning release notes)

**Review by jreback — APPROVED** [review]

**Comment by datapythonista** [comment]
> @datapythonista after this anything else for release (aside from maybe cleaning release notes)

From my side just removing the `Client` classes from SQL backends. With some luck there are no problems with that, and I can get it ready today too, so we can release as soon as you want.

I wouldn't mind removing some backends from this repo, I think that would help with the release too, if we're going to release a package for each backend, which I think it's the plan. But we can leave it for after the release too.

**Comment by jreback** [comment]
> I wouldn't mind removing some backends from this repo, I think that would help with the release too, if we're going to release a package for each backend, which I think it's the plan. But we can leave it for after the release too.

let's leave for after

**Comment by datapythonista** [comment]
@cpcloud mind having a look at this please?

**Comment by cpcloud** [comment]
I am on vacation this week, so I will defer to other reviewers until I am back on the 12th.

**Comment by datapythonista** [comment]
> I am on vacation this week, so I will defer to other reviewers until I am back on the 12th.

Thanks for letting me know. I'll merge and we can follow up if you have any feedback later on.

**Pull request merged**
